pkg/controllers/status/work_status_controller.go

```diff
@@ -171,10 +157,16 @@ func (c *WorkStatusController) syncWorkStatus(key util.QueueKey) error {
 		return fmt.Errorf("invalid key")
 	}
 
-	observedObj, err := helper.GetObjectFromCache(c.RESTMapper, c.InformerManager, fedKey)
+	klog.Infof("Begin to sync status to Work of object(%s)", fedKey.String())
+
+	observedObj, err := c.ResourceInformer.GetObjectFromCache(fedKey)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			return c.handleDeleteEvent(fedKey)
+			executionSpace := names.GenerateExecutionSpaceName(fedKey.Cluster)
+			workName := names.GenerateWorkName(fedKey.Kind, fedKey.Name, fedKey.Namespace)
+			klog.Warningf("Skip reflecting %s(%s/%s) status to Work(%s/%s) for not applied successfully yet.",
+				fedKey.Kind, fedKey.Namespace, fedKey.Name, executionSpace, workName)
+			return nil
```

Comment:
Like this:  
https://github.com/karmada-io/karmada/pull/3808/files/cdbc943859f432dd125ca24c4fe2e3d7e468ae2e#r1267468944  

If the object should be managed by karmada, `execution controller` would take an effect and update the status of `work`, and this event could be received by `bindingstatus controller` to update the status in `binding` and resource on fed-control-plane

Reply:
Shouldn't the work status condition be handled by the execution-controller and the work status manifestStatuses be handled by the work-status-controller, so that they don't conflict with each other?

Reply:
Yes, it seems to be better. That's quite a nice advice.

Reply:
Sorry, after re-read about this part, I find that it seems not necessary to clean up the status in `work`. 

When we try to sync status of an object but the object is missed in cache:  
- If the object is not managed by `karmada`, we don't need to process it.  
- Else, we could wait for the object re-created by `execution-controller`. It could be considered as what we should do when `Applied` condition is `False`. Then the `bindingstatus controller` would take an effect on `binding` and resource on fed-control-plane.  

Or maybe I mis-understood your intention. If so, please explain to me in detail. :)

Reply:
Ok, I agree with you.